### PR TITLE
Add Thrift Installer

### PR DESCRIPTION
**Pull request by uilianries**
Hi!

Following the same idea from Bison, now we can have multiple recipes at same projectt.

Pros:
- Easier to maintain, both are at same project;
- Easier to find. People asked me many time about Protobuf, why there are two projects.

Cons:
- Recipes are more complex, as they use a base class.

### CODE REVIEW

**Comment by Croydon** [comment]
I'm guessing something went wrong with this pull request judging by the look at the commits 🤔 

What I still want to change in the Bison repository (but didn't had the time so far) is to rename the new `.py` files. Same here: Instead of same individual name like `thrift_base.py` and `thrift_installer.py` I would prefer some generic name like `conanfile_base.py` and `conanfile_installer.py`

This makes automation and tools easier.

**Comment by uilianries** [comment]
> I'm guessing something went wrong with this pull request judging by the look at the commits thinking

Yes, I just forgot to rebase my local copy before to start the feature branch :man_facepalming: 

> What I still want to change in the Bison repository (but didn't had the time so far) is to rename the new .py files. Same here: Instead of same individual name like thrift_base.py and thrift_installer.py I would prefer some generic name like conanfile_base.py and conanfile_installer.py

Yes, it sounds better, we can re-use for any project. I'll update it. Thanks

File: test_package_installer/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+class TestPackageConan(ConanFile):
+    settings = "os", "arch"
+
+    def build(self):
+        pass
+
+    def test(self):
+        if tools.cross_building(self.settings):
+            self.output.warn("Skipping test package: Target is incompatible with current arch.")
+        self.run("thrift --version", run_environment=True)
```
Comment (Croydon) [inline test_package_installer/conanfile.py]: this looks wrong to me. If it is NOT cross building we want to execute it.

https://github.com/bincrafters/conan-templates/blob/4637b29f63f53ba21f3279f20d4164332970f8a2/default/test_package/conanfile.py#L16

File: thrift_base.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+import os
+from conans import ConanFile, tools
+
+
+class ThriftBase(ConanFile):
+    version = "0.12.0"
+    description = "Thrift is an associated code generation mechanism for RPC"
+    url = "https://github.com/helmesjo/conan-thrift"
```
Comment (Croydon) [inline thrift_base.py]: This isn't your mistake. I have forgotten to adapt that
```suggestion
    url = "https://github.com/bincrafters/conan-thrift"
```

File: thrift_installer.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        cmake.definitions["BUILD_TUTORIALS"] = False
+        cmake.configure(build_folder=self._build_subfolder)
+        return cmake
+
+    def package_info(self):
+        bin_dir = os.path.join(self.package_folder, "bin")
+        self.env_info.PATH.append(bin_dir)
+
+    def package_id(self):
+        del self.info.settings.compiler
```
Comment (Croydon) [inline thrift_installer.py]:
```suggestion
        del self.info.settings.compiler

```

**Review by Croydon — CHANGES_REQUESTED** [review]
File rename + meta data (not your fault 🙈 )

File: test_package_installer/conanfile.py (hunk 1 above)
Comment (uilianries) [reply test_package_installer/conanfile.py]: you are right! I'll copy from test_package. thanks

Comment (uilianries) [reply test_package_installer/conanfile.py]: done!

File: thrift_base.py (hunk 2 above)
Comment (uilianries) [reply thrift_base.py]: done!

File: thrift_installer.py (hunk 3 above)
Comment (uilianries) [reply thrift_installer.py]: done!

**Comment by Croydon** [comment]
Typo `conan_installer.py` -> `conanfile_installer.py`

**Comment by uilianries** [comment]
This PR is blocked :disappointed: 

thrift_installer requires flex, we have the package for Flex, but the ABI is not compatible because glibc version ...

```
CMake Error at /usr/local/share/cmake-3.12/Modules/FindFLEX.cmake:122 (message):
  Command
  "/home/conan/.conan/data/flex/2.6.4/bincrafters/stable/package/6ae331b72e7e265ca2a3d1d8246faf73aa030238/bin/flex
  --version" failed with output:

  

  
  /home/conan/.conan/data/flex/2.6.4/bincrafters/stable/package/6ae331b72e7e265ca2a3d1d8246faf73aa030238/bin/flex:
  /lib64/libc.so.6: version `GLIBC_2.14' not found (required by
  /home/conan/.conan/data/flex/2.6.4/bincrafters/stable/package/6ae331b72e7e265ca2a3d1d8246faf73aa030238/bin/flex)


Call Stack (most recent call first):
  source_subfolder/compiler/cpp/CMakeLists.txt:32 (find_package)


-- Found FLEX: /home/conan/.conan/data/flex/2.6.4/bincrafters/stable/package/6ae331b72e7e265ca2a3d1d8246faf73aa030238/bin/flex  
CMake Error at /usr/local/share/cmake-3.12/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find BISON (missing: BISON_EXECUTABLE)
Call Stack (most recent call first):
  /usr/local/share/cmake-3.12/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/share/cmake-3.12/Modules/FindBISON.cmake:252 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  source_subfolder/compiler/cpp/CMakeLists.txt:33 (find_package)
```

However, @madebr is doing great job with Flex:

- https://github.com/bincrafters/conan-flex/pull/4
- https://github.com/bincrafters/conan-flex/pull/5

Thus I think as soon as we have the flex_installer, we will be able to release thrift_installer.

Regards!

**Comment by Croydon** [comment]
Flex installer is now available

**Review by Croydon — APPROVED** [review]
LGTM now

Clang 8 fails due to missing prebuilds, but we can fix that later on

**Comment by uilianries** [comment]
Amazing! Thanks @Croydon

**Pull request merged**